### PR TITLE
executor: reduce memory usage for duplicated key update

### PR DESCRIPTION
**Pull request by jackysp**
If the duplicate key is a record key, do not get it from storage, use its value directly. It will reduce some memory usage and reduce one batch get in some scenarios.
PTAL @winkyao

### CODE REVIEW

**Comment by jackysp** [comment]
PTAL @winkyao

File: executor/write.go [hunk 1]
```diff
@@ -1081,20 +1081,27 @@ func (e *InsertExec) initDupOldRowValue(newRows [][]types.Datum) (err error) {
 	for _, keysInRow := range e.uniqueKeysInRows {
 		for _, k := range keysInRow {
 			if val, found := e.dupKeyValues[string(k.key)]; found {
-				var handle int64
-				handle, err = decodeOldHandle(k, val)
-				if err != nil {
-					return errors.Trace(err)
+				if k.isRecordKey {
+					e.dupOldRowValues[string(k.key)] = val
```
Comment (XuHuaiyu) [inline executor/write.go]: extract `string(k.key)` as `key = string(k.key)`

Comment (jackysp) [reply executor/write.go]: It is only a cast for the map key. I think it is not necessary.

Comment (coocood) [reply executor/write.go]: @XuHuaiyu 
If we do this, we need to pay the cost of memory allocation.
Go has an optimization for this.

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**